packages/react-components/react-virtualizer/src/components/Virtualizer/Virtualizer.types.ts

```diff
@@ -133,16 +133,6 @@ export type VirtualizerConfigProps = {
    * @param index - the index of the requested size's child
    */
   getItemSize?: (index: number) => number;
-
-  /**
-   * Notify users of index changes
-   */
-  onUpdateIndex?: (index: number, prevIndex: number) => void;
```

Comment:
why did you remove these change callbacks? they would have also achieved the same result without using context to override control from the virtualizer itself.

I feel the use of context is a bit odd. From what I understand you need to know the current index to be able to calculate buffer sizes for the dynamic scroll view. However, the context control seems to be breaking encapsulation because the virtualizer scroll view never actually needs to control the value of the current index - that responsibility belongs to the virtualizer.

The scroll view only reads the current index, it never actually sets it, which is what makes me think the encapsulation is being broken here.

Since you only react to the resize observer updates and not the actual render you could do something like this (this is pseudocode to show the approach):

```tsx
const currentIndexRef = React.useRef(-1);

const onUpdateIndex = (index,) => {
  currentIndexRef.current = index;
}

new ResizeObserver(() => {
  // do something with currentIndexRef here
})

<Virtualizer onUpdateIndex={onUpdateIndex} />
```